packages/FormElement/src/FormElement.js

```diff
@@ -125,23 +131,30 @@ function FormElement(props) {
     });
   }
   return (
-    <div css={formElementStyles} isInline={isInline} size={size} isDisabled={isDisabled} {...moreProps}>
+    <sc.FormElement
+      as={hasFieldSet ? "fieldset" : "div"}
+      isInline={isInline}
+      size={size}
+      isDisabled={isDisabled}
+      {...moreProps}
+    >
       <Label
-        hasOptionalLabel={hasOptionalLabel}
+        hasOptionalLabel={hasRequiredLabel ? false : hasOptionalLabel}
```

Comment:
Changed the logic here so that `required` label takes priority and will not show the `optional` field if it is `hasRequiredLabel` is true